src/main/kotlin/com/github/gradle/node/NodePlugin.kt

```diff
@@ -6,58 +6,57 @@ import com.github.gradle.node.npm.task.NpmTask
 import com.github.gradle.node.npm.task.NpxTask
 import com.github.gradle.node.task.NodeSetupTask
 import com.github.gradle.node.task.NodeTask
-import com.github.gradle.node.variant.VariantBuilder
 import com.github.gradle.node.yarn.task.YarnInstallTask
 import com.github.gradle.node.yarn.task.YarnSetupTask
 import com.github.gradle.node.yarn.task.YarnTask
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.register
 
 class NodePlugin : Plugin<Project> {
     private lateinit var project: Project
-    private lateinit var nodeExtension: NodeExtension
 
     override fun apply(project: Project) {
         this.project = project
-        this.nodeExtension = NodeExtension.create(project)
+        val nodeExtension = NodeExtension.create(project)
         addGlobalTypes()
         addTasks()
         addNpmRule()
         addYarnRule()
-        this.project.afterEvaluate {
-            nodeExtension.variant = VariantBuilder().build(nodeExtension)
+        project.afterEvaluate {
+            nodeExtension.finalize()
```

Comment:
Do we need to finalise the values?

Reply:
Ok given `finalizeOnRead` was introduced in 6.1 we probably do :'(

Reply:
Maybe we could adapt this behavior according to the Gradle version?

I don't know whether it is really important.